### ISSUE TITLE
`sample()` with `temperature=0` does not produce greedy decoding when `top_p > 0`

### ISSUE THREAD

**Issue by levguy** [Lightning-AI/litgpt#2238]
## Summary

Setting `temperature=0` is universally understood as greedy decoding — always pick the most probable token. However, `sample()` in `litgpt/generate/base.py` does **not** produce greedy decoding when `temperature=0` and `top_p` is at its default value of `1.0`. Instead, the model samples randomly from the full softmax distribution, making `temperature=0` completely ineffective.

## Root Cause

The branching condition is:

```python
if temperature > 0.0 or top_p > 0.0:
```

When `temperature=0` and `top_p=1.0` (the default), this evaluates to `False or True` → `True`, so the code enters the probabilistic sampling branch. Inside that branch, temperature scaling is correctly skipped (since `temperature > 0.0` is false), but the logits are still passed through `softmax` and sampled via `multinomial`. The `argmax` (greedy) path is only reached when **both** `temperature == 0` **and** `top_p == 0`.

The net effect is that `temperature=0, top_p=1.0` behaves identically to `temperature=1.0, top_p=1.0` — the temperature setting has zero effect on the output.

## Reproduction

```python
import torch
from litgpt.generate.base import sample

logits = torch.tensor([[[0.5, -1.2, 3.1, 0.8, -0.3, 2.7, -0.9, 1.4]]])

expected = torch.argmax(logits[0, -1], dim=-1).item()  # 2 (the 3.1)

results = [sample(logits, temperature=0.0, top_p=1.0).item() for _ in range(10)]
print(f"Expected (argmax): {expected}")
print(f"Got: {results}")
# Bug: results should be [2, 2, 2, 2, 2, 2, 2, 2, 2, 2]
# but instead vary randomly because temperature=0 is ignored
```

## Suggested Fix

Change `or` to `and`:

```python
if temperature > 0.0 and top_p > 0.0:
```

This ensures:
- `temperature=0` → always greedy (argmax), regardless of `top_p`
- `top_p=0` → always greedy (argmax), regardless of `temperature`
- Both positive → standard probabilistic sampling

### What operating system are you using?

Linux

**Comment by CrepuscularIRIS** [comment]
I can reproduce this — with `temperature=0.0` and `top_p=1.0` (the default), the `or` condition at the branch guard sends execution into the probabilistic sampling path instead of the `argmax` (greedy) path.

I'll submit a fix that changes the guard from `or` to `and`, so `temperature=0` always produces greedy decoding regardless of the `top_p` value. Will include a regression test.

**Comment by levguy** [comment]
Thank you for the quick fix!
